altair/view.py

```diff
@@ -1,116 +1,239 @@
-
-    for shelf in shelves:
-        if getattr(enc, shelf, None) is None:
-            setattr(enc, shelf, _shelf_types[shelf](''))
-        factory = _shelf_view_factories[shelf]
-        view = factory(getattr(enc, shelf), columns, types)
-        children.append(view)
-    
-    vbox.children = children
-    return vbox
-
-
-def create_viz_view(viz, shelves=_shelves):
-    columns, types = find_columns_and_types(viz.data)
-    vbox = w.VBox()
-    hbox = w.HBox()
-    heading = w.HTML('Mark:', width='221px', height='32px')
-    dd = create_dropdown_view(viz, 'marktype')
-    hbox.children = [heading, dd]
-    if not hasattr(viz, 'encoding'):
-        setattr(viz, 'encoding', Encoding())
-    encoding = create_encoding_view(viz.encoding, columns, types, shelves=shelves)
-    vbox.children = [hbox, encoding]
-    return vbox
-
-
-# Not exposing this yet because of bugs in the renderers that break this.
-
-# def create_render_view(v, shelves=_shelves):
-#     render_button = w.Button(description='Render')
-#     output = w.Output()
-#     render_box = w.VBox()
-#     render_box.children = [render_button, output]
-#
-#     viz_view = create_viz_view(v)
-#     hbox = w.HBox()
-#     hbox.children = [viz_view, render_box]
-#     def on_render(*args):
-#         with output:
-#             v.render()
-#     render_button.on_click(on_render)
-#     return hbox
+class AltairWidget(object):
+    _shelves = ('x', 'y', 'row', 'col', 'shape', 'color', 'size', 'detail', 'text')
 
+    _shelf_types = {
+        'x': X,
+        'y': Y,
+        'row': Row,
+        'col': Col,
+        'shape': Shape,
+        'color': Color,
+        'detail': Detail,
+        'text': Text,
+        'size': Size
+    }
+
+    def __init__(self, viz, shelves=_shelves, renderer='lightning'):
+        use_renderer(renderer)
+
+        self.viz = viz
+        self.shelves = shelves
+
+        self._shelf_view_factories = {
+            'x': self._create_shelf_view,
+            'y': self._create_shelf_view,
+            'row': self._create_shelf_view,
+            'col': self._create_shelf_view,
+            'shape': self._create_shelf_view,
+            'color': self._create_shelf_view,
+            'detail': self._create_shelf_view,
+            'text': self._create_shelf_view,
+            'size': self._create_shelf_view
+        }
+
+        self.widget = w.HBox()
+
+        # Create output area
+        self.output = w.HBox()
+        self.output.children = []
+
+        # Create widget
+        self.controls = self._create_left_pane()
+
+        self.widget.children = [self.controls, self.output]
+
+    def render(self):
+        return self.widget
+
+    def on_render_viz(self, *args):
+        self._clear_output()
+
+        to_display = w.Textarea(value=str(self.viz))
+
+        # TODO: how to display properly?
+        # to_display = w.Output()
+        # with to_display:
+        #     self.viz.render()
```

Comment:
ping @mathisonian @freeman-lab 

If you were to comment out line 65 and uncomment 61-63, you should see the lightning rendered graph, but instead you only get blank space. I see no errors, though, so I don't know why there's no display of anything. I wonder if this is the same thing @ellisonbg saw.